CMakeLists.txt

```diff
@@ -149,14 +158,25 @@ if(CLI11_SINGLE_FILE)
     target_include_directories(CLI11_SINGLE INTERFACE "${CMAKE_CURRENT_BINARY_DIR}/include/")
 endif()
 
-cmake_dependent_option(CLI11_SINGLE_FILE_TESTS "Duplicate all the tests for a single file build" OFF "CLI11_SINGLE_FILE" OFF)
+cmake_dependent_option(CLI11_SINGLE_FILE_TESTS
+    "Duplicate all the tests for a single file build"
+    OFF
+    "CLI11_SINGLE_FILE"
+    OFF)
 
-cmake_dependent_option(CLI11_TESTING "Build the tests and add them" ON "CUR_PROJ" OFF)
-if(DEFINED BUILD_TESTING)
-    cmake_dependent_option(CLI11_TESTING "" ON "BUILD_TESTING" OFF)
-    message(STATUS "BUILD_TESTING is defined and it supersedes CLI11_TESTING. Has forced to ${CLI11_TESTING}")
+cmake_dependent_option(CLI11_TESTING
+    "Build the tests and add them (BUILD_TESTING also works if this is the main project)"
+    ON
+    "CUR_PROJ"
+    OFF)
+
+if(DEFINED BUILD_TESTING AND CUR_PROJ)
+    set(CLI11_TESTING_INTERNAL ON)
+else()
+    set(CLI11_TESTING_INTERNAL "${CLI11_TESTING}")
```

Comment:
These conditions don't seem quite right.  
If BUILD_TESTING is defined to OFF in a main project then testing is turned ON regardless, and I don't necessarily see a way to turn if off, and if BUILD_TESTING is defined and it is a main project then CLI11_TESTING is ignored completely.

Reply:
Thanks, I think that was a copy-and-paste mistake. Maybe defining both CLI11_TESTING and BUILD_TESTING in a main project should be an error? (At least if they differ)

Reply:
I think my preference would be an AND condition at least for the case of a main project.

Reply:
`CLI11_TESTING` will be respected if it is defined in all cases, otherwise `BUILD_TESTING` is defined and set to ON if this is a main project. I think that's probably the best way to go, especially since I'm probably going to recommend `BUILD_TESTING` + main project only for Modern CMake.